ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+            the 'input_url' parameter.
+        """
+
+        file_list = glob.glob(kwargs['input_url'])
+        file_list.sort()
+
+        # EventSource can not handle file wild cards as input_url
+        # To overcome this we substitute the input_url with first file matching
+        # the specified file mask.
+        del kwargs['input_url']
+        super().__init__(config=config, tool=tool, input_url=file_list[0], **kwargs)
+
+        # Retrieving the list of run numbers corresponding to the data files
+        run_numbers = list(map(self._get_run_number, file_list))
+        self.run_numbers = np.unique(run_numbers)
+
+        # # Setting up the current run with the first run present in the data
+        # self.current_run = self._set_active_run(run_number=0)
+        self.current_run = None
+        
+        # MAGIC telescope positions in m wrt. to the center of CTA simulations
+        self.magic_tel_positions = {
+            1: [-27.24, -146.66, 50.00] * u.m,
+            2: [-96.44, -96.77, 51.00] * u.m
+        }
+        # MAGIC telescope description
+        optics = OpticsDescription.from_name('MAGIC')
+        geom = CameraGeometry.from_name('MAGICCam')
+        self.magic_tel_description = TelescopeDescription(optics=optics, camera=geom)
+        self.magic_tel_descriptions = {1: self.magic_tel_description, 2: self.magic_tel_description}
+        self.magic_subarray = SubarrayDescription('MAGIC', self.magic_tel_positions, self.magic_tel_descriptions)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return
+
+    @staticmethod
+    def is_compatible(file_mask):
+        """
+        This method checks if the specified file mask corresponds
+        to MAGIC data files. The result will be True only if all
+        the files are of ROOT format and contain an 'Events' tree.
+
+        Parameters
+        ----------
+        file_mask: str
+            A file mask to check
+
+        Returns
+        -------
+        bool:
+            True if the masked files are MAGIC data runs, False otherwise.
+
+        """
+
+        is_magic_root_file = True
+
+        file_list = glob.glob(file_mask)
+
+        for file_path in file_list:
+            try:
+                with uproot.open(file_path) as input_data:
+                    if 'Events' not in input_data:
+                        is_magic_root_file = False
+            except ValueError:
+                # uproot raises ValueError if the file is not a ROOT file
+                is_magic_root_file = False
+                pass
+
+        return is_magic_root_file
+
+    @staticmethod
+    def _get_run_number(file_name):
+        """
+        This internal method extracts the run number from
+        the specified file name.
+
+        Parameters
+        ----------
+        file_name: str
+            A file name to process.
+
+        Returns
+        -------
+        int:
+            A run number of the file.
+        """
```

Comment:
What if the original MAGIC file has been renamed e.g. for testing purposes from its original file name to `test_magic_file.root`. I assume in that case this method would raise an exception. 
Would it be possible to fail gracefully in this case and return maybe `nan` or `-1` or `"unkown"` or something like that?
So that one can read events from a file, even if the run-number cannot be found out?

Reply:
There has to be at least two files - one for Magic-I and another for Magic-II; these are normally linked to each other via the run number. So if there run number is not there, proper data reading can not be guaranteed anyway.
if the run number is set to nan/-1/unknown etc, the corresponding files can not be found further in _set_active_run(). Maybe a specially tailored exception would be a better option here? Say, like so:
 ```
try:
    run_number = int(parsed_info[0])
except IndexError:
    raise IndexError('Can not identify the run number of the file {:s}'.format(file_name))
```

Reply:
Thank you for the clarification. Indeed after reading more of this code I understood how big the influence of the files path to the reader is. So one cannot simply rename a MAGIC file and assume it can still be read, the name (or complete path?) is a part of the file format as it contains important information as you pointed out. 

So I would like to mark this comment as **resolved**, but github does not have this feature.